src/components/ui/Toast/Toast.tsx

```diff
@@ -1,35 +1,44 @@
-import Button from 'src/components/ui/Button'
-import { useEffect, useRef } from 'react'
+import { useEffect, useRef, useState } from 'react'
 import { useUI } from 'src/sdk/ui'
-
-const state = {
-  ERROR: 'bg-red',
-  WARNING: 'bg-yellow',
-  INFO: 'bg-green',
-}
+import Icon from 'src/components/ui/Icon'
 
 function Toast() {
   const { toasts, popToast } = useUI()
   const toast = toasts[toasts.length - 1]
   const ref = useRef<NodeJS.Timeout[]>([])
 
+  const [fade, setFade] = useState<'in' | 'out'>()
+
+  useEffect(() => {
+    toast && setFade('in')
+  }, [toast])
+
   useEffect(() => {
-    const id = setTimeout(() => {
-      popToast()
-    }, 2e3)
+    if (fade === 'in') {
+      const id = setTimeout(() => setFade('out'), 2e3)
```

Comment:
does make sense to clear the Timeout inside the [cleanup function](https://reactjs.org/docs/hooks-effect.html#:~:text=Why%20did%20we%20return%20a%20function%20from%20our%20effect%3F) return of `useEffect`? something like:
```js
return () => {
  clearTimeout(id);
};
```

Reply:
humm, I think it makes sense, let me put this cleanup function and check if there's some interference on the behavior